CHANGES.md

```diff
@@ -1,3 +1,144 @@
+## Version 3.0.0-RC4 (August 25, 2019)
+
+Last release before `3.0.0` which will be released as soon as Cats-Effect 2.0.0 is available. At the time of writing release notes,
+it is only waiting on Cats 2.0.0 which is about to release next RC which will become stable version if no critical issues are discovered.
+
+Any other development for this release is now frozen except if we discover critical bugs like memory leaks. All other
+changes will target `3.1.0`. 
+
+`3.0.0` will be binary and source compatible with `3.0.0-RC4` for Scala 2.12. Monix itself will be also binary compatible
+for 2.11 but it will have a dependency on Cats-Effect which is not. [See Cats-Effect release notes](https://github.com/typelevel/cats-effect/releases/tag/v2.0.0-RC1) for more details.
+
+We're very sorry for such a long wait. We wish we could release `3.0.0` already but if we released now, we would have to 
+release `4.0.0` for Cats-Effect 2.0 because of 2.11 incompatibility there so we decided to hold on. 
+We hope you can understand and wait a little longer. It was a rough road but currently we have 3 active maintainers instead
+of just Alex so I hope it can give you a confidence for the future! :)
+Note that Cats-Effect 2.0 is very small release and mostly targeted at support for Scala 2.13 and bug fixes - the upgrade should be limited to bumping version without
+changing a single line of code.
+
+This release depends on Cats-Effect 1.4.0 and Cats 1.6.1
+
+This release is binary compatible with `3.0.0-RC3` on Scala 2.12.
+On Scala 2.11 there is an incompatibility caused by introduction of `Scheduler.features`:
+
+```
+exclude[ReversedMissingMethodProblem]("monix.execution.Scheduler.features")
+```
+
+### Main changes
+
+#### Local
+
+There are several fixes related to `Local` usage.
+- Using `TracingScheduler` will also automatically enable local context propagation in `Task` so
+running it with `runToFutureOpt` or setting env variable is no longer a necessity.
+- `Local.isolate` has a overload for isolating `Future` which is safer than regular `Local.isolate`.
+
+The `Local` model since `3.0.0-RC3` shares by default. To isolate modifications of `Local` by
+other concurrently running `Future`, you have to call `Local.isolate`. 
+
+In case of `Task`, there is a `TaskLocal.isolate` version. It is automatically called whenever you run `Task`
+so if your use case is setting `correlationId` or similar, it probably won't require any explicit isolation because
+your HTTP library will most likely run the `Task` per request.
+
+NOTE: We are still looking for improvements for `Local` and it is likely that it will see additional changes in subsequent releases.
```

Comment:
Well I hope the current design is solid enough that we shouldn't break binary compatibility for it. Otherwise we're screwed 🙂

But it's solid enough so I have my hopes up.

Reply:
I hope so too but if we discover possible improvements, such as doing it without support in `Task` run-loop or figure out a sensible way to auto-isolate in parallel operators without breaking `fa.start.flatMap(_.join) <-> fa` then I think it could be worth considering

Do you think we should remove/reword this note?

Reply:
Yes. Delete this paragraph too.

Reply:
Ok, removed :)